plugins/reach/lib/model/EntryVendorTask.php

```diff
@@ -235,9 +240,83 @@ public function getServiceFeature()
 		return $this->getFromCustomData(self::CUSTOM_DATA_SERVICE_FEATURE);
 	}
 
-	/* (non-PHPdoc)
- 	 * @see BaseEntryVendorTask::preSave()
- 	 */
+	protected function calculateNextBusinessDay()
```

Comment:
This function should be generic - calculate next day from timestamp and not from now.
Add timestamp in the API call.